spec/models/metric_spec.rb

```diff
@@ -67,17 +67,17 @@
           expect(MiqQueue.count).to eq(47)
 
           expected_targets = Metric::Targets.capture_targets
-          expected = expected_targets.collect do |t|
+          expected = expected_targets.flat_map do |t|
             # Storage is hourly only
             # Non-storage historical is expecting 7 days back, plus partial day = 8
-            t.kind_of?(Storage) ? [t, "hourly"] : [[t, "realtime"], [t, "historical"] * 8]
-          end.flatten
+            t.kind_of?(Storage) ? [t, "hourly"] : [[t, "realtime"], [t, "historical"] * 8].flatten
```

Comment:
I don't think you want flatten here.  You want to keep the time and intervals paired together.  This completely flatten them, and then match_array will not do what is expected.  However, the original code completely flattened because it relied on order, so it was ok.  If you are not relying on order, I think you can't use `[t, "historical"] * 8` anymore.  You'd probably have to do `[[t, "historical"]] * 8` instead.